tests/kafkatest/tests/core/downgrade_test.py

```diff
@@ -21,71 +21,84 @@
             self.kafka.start_node(node)
 
+    def setup_services(self, kafka_version, compression_types, security_protocol):
+        self.create_zookeeper()
+        self.zk.start()
+
+        self.create_kafka(num_nodes=3,
+                          security_protocol=security_protocol,
+                          interbroker_security_protocol=security_protocol,
+                          version=kafka_version)
+        self.kafka.start()
+
+        self.create_producer(log_level="DEBUG",
+                             compression_types=compression_types,
+                             version=kafka_version)
+        self.producer.start()
+
+        self.create_consumer(log_level="DEBUG",
+                             version=kafka_version)
+        self.consumer.start()
+
     @cluster(num_nodes=7)
-    @parametrize(kafka_version=str(LATEST_2_3), compression_types=["none"])
-    @parametrize(kafka_version=str(LATEST_2_3), compression_types=["zstd"])
-    @parametrize(kafka_version=str(LATEST_2_2), compression_types=["none"])
-    @parametrize(kafka_version=str(LATEST_2_2), compression_types=["zstd"])
-    @parametrize(kafka_version=str(LATEST_2_1), compression_types=["none"])
-    @parametrize(kafka_version=str(LATEST_2_1), compression_types=["lz4"])
-    @parametrize(kafka_version=str(LATEST_2_0), compression_types=["none"])
-    @parametrize(kafka_version=str(LATEST_2_0), compression_types=["snappy"])
-    @parametrize(kafka_version=str(LATEST_1_1), compression_types=["none"])
-    @parametrize(kafka_version=str(LATEST_1_1), compression_types=["lz4"])
-    @parametrize(kafka_version=str(LATEST_1_0), compression_types=["none"])
-    @parametrize(kafka_version=str(LATEST_1_0), compression_types=["snappy"])
-    @parametrize(kafka_version=str(LATEST_0_11_0), compression_types=["none"])
-    @parametrize(kafka_version=str(LATEST_0_11_0), compression_types=["lz4"])
-    @parametrize(kafka_version=str(LATEST_0_10_2), compression_types=["none"])
-    @parametrize(kafka_version=str(LATEST_0_10_2), compression_types=["lz4"])
-    @parametrize(kafka_version=str(LATEST_0_10_1), compression_types=["none"])
-    @parametrize(kafka_version=str(LATEST_0_10_1), compression_types=["gzip"])
-    @parametrize(kafka_version=str(LATEST_0_10_0), compression_types=["none"])
-    @parametrize(kafka_version=str(LATEST_0_10_0), compression_types=["lz4"])
-    @parametrize(kafka_version=str(LATEST_0_9), compression_types=["none"], security_protocol="SASL_SSL")
-    @parametrize(kafka_version=str(LATEST_0_9), compression_types=["snappy"])
-    @parametrize(kafka_version=str(LATEST_0_9), compression_types=["lz4"])
-    @parametrize(kafka_version=str(LATEST_0_8_2), compression_types=["none"])
-    @parametrize(kafka_version=str(LATEST_0_8_2), compression_types=["snappy"])
-    def test_upgrade_and_downgrade(self, kafka_version, compression_types, security_protocol="PLAINTEXT"):
+    @parametrize(version=str(LATEST_2_3), compression_types=["none"])
```

Comment:
Could some of these be replaced with `@matrix`?

Reply:
Possibly. Let me consider it.